Price decrease

DonatasPuc opened the pull request:
Implemented price decrease. I don't know if it is the optimal solution, but currently the client can pick price decrease type (either "AMOUNT" or "PERCENT") by which current price is calculated and returned in a FoodResponse dto.

audrius-savickas commented on backend/Persistence/DB/Data/Test/Foods.json:
```diff
@@ -1,150 +1,218 @@
+    "IdRestaurant": "0",
+    "TypesOfFood": [],
+    "StartDecreasingAt": "2021-11-04T11:47:04.9955968+02:00",
+    "DecreaseType": "PERCENT",
```
I think we should use enums for `DecreaseType`, for example 0 for percent type, and 1 for value type

audrius-savickas commented on backend/Persistence/DB/Data/Test/Foods.json:
```diff
@@ -164,7 +232,30 @@
       }
     ],
+    "StartDecreasingAt": "2021-11-04T11:47:04.9955968+02:00",
+    "DecreaseType": "PERCENT",
```
foods.json should have foods with decrease type of value as well, not only percent like right now

audrius-savickas commented on backend/Persistence/DB/Data/Test/Foods.json:
```diff
@@ -164,7 +232,30 @@
+      }
+    ],
+    "StartDecreasingAt": "2021-11-04T09:59:21.56Z",
+    "DecreaseType": "string",
```
string?

audrius-savickas commented on backend/Domain/Entities/Food.cs:
```diff
@@ -5,28 +5,99 @@ namespace Domain.Entities
+        public double PercentPerInterval { get; set; }
+
+        private const string AMOUNT_STR = "AMOUNT";
+        private const string PERCENT_STR = "PERCENT";
```
move these to enum

audrius-savickas reviewed (changes requested):
I'm not sure if this exception throwing in `Food.cs` constructor is the most effective, I think you can improve the readability a bit

DonatasPuc commented:
Changed DecreaseType to enum. Added validation to api's for validating DecreaseType enum value. Improved data in joods.json.

Refactored Food constructor, but I still don't know if it's ok.

Also, where should I document what enum values mean in the web api?

JesusGonzalezA commented:
> Also, where should I document what enum values mean in the web API?

We should, but we would have to import new packages or create new classes just to alter the documentation UI. I believe it takes time and it does not really give us a huge increase of value. For me it is correct leaving it alone for now
